### PR TITLE
LITE-21209: Added backoff retry reconnection and user agent

### PR DESCRIPTION
**Pull request by gab832**

### CODE REVIEW

File: connect/eaas/config.py [hunk 1]
```diff
@@ -89,8 +92,29 @@ def get_ws_url(self):
     def get_api_url(self):
         return f'https://{self.env["api_address"]}/public/v1'
 
+    def get_version(self):
```
Comment (ffaraoneim) [inline connect/eaas/config.py:95]: This function should be moved outside config.

File: connect/eaas/config.py [hunk 2]
```diff
@@ -89,8 +92,29 @@ def get_ws_url(self):
...
+            return '0.0.0'
+
+    def get_user_agent(self):
+        version = self.get_version()
+        pimpl = platform.python_implementation()
+        pver = platform.python_version()
+        sysname = platform.system()
+        sysver = platform.release()
+        ua = (
+            f'connect-extension-runner/{version} {self.env["environment_id"]}/'
```
Comment (ffaraoneim) [inline connect/eaas/config.py:108]: Personally I would prefer that environment and instance to be at the end of the user agent string.

File: connect/eaas/config.py [hunk 3]
```diff
@@ -89,8 +92,29 @@ def get_ws_url(self):
...
+            f'{self.env["instance_id"]} {pimpl}/{pver} {sysname}/{sysver}'
+        )
+        return {'User-Agent': ua}
+
     def get_headers(self):
-        return (('Authorization', self.api_key),)
+        return (
+            ('Authorization', self.api_key),
+            ('User-Agent', self.get_user_agent()),
```
Comment (ffaraoneim) [inline connect/eaas/config.py:116]: Customized User-Agent is for ConnectClient instances (both sync and async) not for the websocket.

File: connect/eaas/worker.py [hunk 4]
```diff
@@ -36,6 +46,31 @@
...
+    return MAX_RETRY_DELAY_TIME_SECONDS
+
+
+def _on_communication_backoff(details):
+    ordinal = {
+        1: 'st',
+        2: 'nd',
+        3: 'rd',
+    }
+    ordinal_attempt = ordinal.get(details["tries"], 'th')
```
Comment (ffaraoneim) [inline connect/eaas/worker.py:67]: Could you please replace double quotes for single ones ?

**Review by ffaraoneim — CHANGES_REQUESTED** [review]

**Review by ffaraoneim — APPROVED** [review]

**Review by marcserrat — APPROVED** [review]

**Pull request merged**